internal/diff_str.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import type { DiffResult } from "./_types.ts";
+import { _internals } from "./diff.ts";
```

Comment:
Is `_internals` object necessary? It seems only used for accessing to `diff` function like `_internals.diff()`. Is it possible to simply import `diff` instead of `_internals`?

Reply:
It's needed to mock a throw in `diff()` to cater for `[Cannot display]` output in tests. If #4700 and #4701 go ahead, it'll no longer be needed.

Reply:
Ah, ok. Then let's keep it for now